index.html

```diff
@@ -2449,6 +2695,254 @@ <h3 id=new-session><dfn data-lt="new sessions|creating a new session">New Sessio
+
+ <li><p>If the <a>maximum active sessions</a> is equal to
+  the length of the list of <a>active sessions</a>,
+  return <a>error</a> with <a>error code</a> <a>session not created</a>.
+
+ <li><p>Let <var>capabilities</var> be the result
+   of <a>trying</a> to <a>process capabilities</a>
+   with <var>parameters</var> as an argument.
+
+ <li><p>If <var>capabilities</var> is <a><code>null</code></a>,
+   return <a>error</a> with <a>error code</a> <a>session not created</a>.
+
+ <li><p>Let <var>session creation job</var> be a new JSON <a>Object</a>
+  with the following properties:
+
+  <dl>
+    <dt>"<code>sessionCreationJobId</code>"
+    <dd>The result of <a>generating a UUID</a>.
+
+    <dt>"<code>status</code>"
+    <dd>The String constant "<code>queuing</code>".
+
+    <dt>"<code>capabilitiesRequest</code>"
+    <dd>The result of <a>getting the property</a> named <var>capabilities</var> from <var>parameters</var>.
+
+    <dt>"<code>timeout</code>"
+    <dd><var>timeout</var>’s value.
+
+    <dt>"<code>requestedTime</code>"
+    <dd><var>session creation requested time</var>’s value.
+
+    <dt>"<code>events</code>"
+    <dd>An empty JSON <a>List</a>.
+
+     <dt>"<code>data</code>"
+     <dd>A JSON <a>Object</a> with implementation-specific data.
+   </dl>
+
+  <li>Add the <var>session creation job</var> to the list of
+   <a>session creation jobs</a>.
+</ol>
+
+</section> <!-- /Async New Session -->
+
+<section>
+<h3><dfn>Async New Session Status</dfn></h3>
+
+<table class="simple jsoncommand">
+ <tr>
+  <th>HTTP Method</th>
+  <th>URI Template</th>
+ </tr>
+ <tr>
+  <td>GET</td>
+  <td>/session/async/{<var>session creation job id</var>}{/<var>event</var>}</td>
+ </tr>
+</table>
+
+<p>The <a>remote end steps</a> are:
+
+<ol>
+  <ol>
+    <li><p>Let <var>session creation job</var> be the job in <a>session creation jobs</a>
+     with a <a>session creation job id</a> matching <var>session creation job id</var>.
+
+    <li><p>If <var>session creation job</var> is <a><code>null</code></a>,
+     return <a>error</a> with <a>error code</a> <a>invalid session creation job id</a>.
+
+    <li><p>Let <var>body</var> be a <a>JSON clone</a> of <var>session creation job</var>.
+
+    <li><p>If <var>event</var> is defined:
+
+     <ol>
+      <li><p>Let <var>filtered events</var> be an empty JSON <a>List</a>.
+
+      <li><p>Let <var>found</var> be false.
+
+      <li><p>Let <var>events</var> be the value of <a>getting a property</a> named <var>events</var> from <var>body</var>.
+
+      <li><p>For each <var>event object</var> in <var>events</var> in reverse order:
+
+      <ol>
+       <li><p>If <var>found</var> is true, then break.
+
+       <li><p>Let <var>event id</var> be the value of <a>getting a property</a> named <var>eventId</var> from <var>event object</var>.
+
+       <li><p>If <var>event</var> equals <var>event id</var>,
+        set <var>found</var> to true.
+
+       <li><p>Prepend <var>event object</var> to <var>filtered events</var>.
```

Comment:
We only ever collect a single event from this loop. Why do we maintain the list?